verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -42,25 +51,177 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
     return status;
   }
 
-  // TODO(fangism): parse a subcommand-specific flag for replacement option
-  // See documentation for 'replacement' arg of the library function:
-  verilog::StripVerilogComments(source_contents, &outs, ' ');
+  char replace_char;
+  if (args.size() == 1) {
+    replace_char = ' ';
+  } else if (args.size() == 2) {
+    absl::string_view replace_str(args[1]);
+    if (replace_str.empty())
+      replace_char = '\0';
+    else if (replace_str.length() == 1)
+      replace_char = replace_str[0];
+    else
+      return absl::InvalidArgumentError(
+          "Replacement must be a single character.");
+  } else {
+    return absl::InvalidArgumentError("Too many arguments.");
+  }
+
+  verilog::StripVerilogComments(source_contents, &outs, replace_char);
 
   return absl::OkStatus();
 }
 
-static const std::pair<absl::string_view, SubcommandEntry> kCommands[] = {
-    {"strip-comments",  //
-     {&StripComments,   //
-      R"(strip-comments file
+static absl::Status PreprocessSingleFile(const char* source_file, std::istream&,
+                                         std::ostream& outs, std::ostream&) {
+  std::string source_contents;
+  if (auto status = verible::file::GetContents(source_file, &source_contents);
+      !status.ok()) {
+    std::cerr << "ERROR: passed file can't be open\n.";
```

Comment:
We have more information like the filename and the status which actually contains the exact problem, so we can print that directly:

```
  message_stream << filename << ": " << status;
```